### PR TITLE
Fix styled prompts on Windows

### LINKED ISSUE #27: Styled prompts on Windows behave incorrectly
Windows currently displays styled prompts *correctly* but places the cursor much too far to the right. This is because `rustyline::Editor::readline` does not expected an ANSI escaped string. ANSI escapes need to be added in the `Highlighter::highlight_prompt` implementation.

### PR DESCRIPTION
**Pull request by adam-mcdaniel**
On windows, colored prompts are displayed correctly, but the cursor is too far to the right. This is because `readline` is intended to be called without ANSI escaped strings. To correct this, I simply store the styled/colored string in the `DuneHelper` structure for `rustyline::Editor`, strip the ANSI escape sequences from the prompt when passing to `rustyline::Editor::readline`, and then return the original styled string in the `DuneHelper::highlight_prompt` implementation of `rustyline::Highlighter`.

In other words, this PR fixes the incorrect usage of `rustyline`.

### CODE REVIEW

**Comment by adam-mcdaniel** [comment]
I will check that this runs on windows correctly before confirming the merge.
